### PR TITLE
[FEATURE] Add flush errors button to backend module

### PR DESCRIPTION
**Pull request by markusguenther**
It is not possible to reset the error currently.

This pull request add an button to the backend module to flush the error field.

### CODE REVIEW

File: Classes/Backend/SolrModule/IndexQueueModuleController.php [hunk 1]
```diff
@@ -67,7 +67,7 @@ public function indexAction() {
 	 * @return void
 	 */
 	public function initializeIndexQueueAction() {
-		$initializedIndexingConfigurations = array();
+		$initializedIndexingConfigurations = [];
```
Comment (irnnr) [inline Classes/Backend/SolrModule/IndexQueueModuleController.php]: Please revert these unrelated changes. Also, array short syntax is only available with PHP 5.4+, but CMS 6.2 only requires a minimum version of 5.3.7.

File: Classes/Backend/SolrModule/IndexQueueModuleController.php [hunk 2]
```diff
@@ -114,6 +114,32 @@ public function initializeIndexQueueAction() {
...
 
 	/**
+	 * Removes all errors in the index queue list. So that the items can be indexed again.
+	 *
+	 * @return void
+	 */
+	public function flushLogErrorsAction() {
+		/** @var DatabaseConnection $database */
+		$database = $GLOBALS['TYPO3_DB'];
+		$flushResult = $database->exec_UPDATEquery('tx_solr_indexqueue_item', 'errors NOT LIKE ""', ['errors' => '']);
```
Comment (irnnr) [inline Classes/Backend/SolrModule/IndexQueueModuleController.php]: same her, please don't use array short syntax (although I agree it's a lot nicer)

File: Classes/Backend/SolrModule/IndexQueueModuleController.php [hunk 3]
```diff
@@ -114,6 +114,32 @@ public function initializeIndexQueueAction() {
...
+
+		$message = 'All errors has been flushed';
```
Comment (irnnr) [inline Classes/Backend/SolrModule/IndexQueueModuleController.php]: Let's go with this: "All errors *have* been reset."

**Comment by irnnr** [comment]
Great change, just a few items that prevent merging it.

File: Classes/Backend/SolrModule/IndexQueueModuleController.php (hunk 3 above)
Comment (markusguenther) [inline Classes/Backend/SolrModule/IndexQueueModuleController.php]: To be honest i did not used locallang labels because the whole module is not using it ;). But it would be nice for sure.

**Comment by markusguenther** [comment]
ok some adjustments and the localisation of the index queue module.
If this is ok for you i can localize the other controllers later in a different pull request.
